iotileship/iotile/ship/recipe.py

```diff
@@ -88,7 +89,33 @@ def _process_yaml(cls, yamlfile):
             info = yaml.load(f)
             return info
 
-    def prepare(self, variables=None):
+    def _complete_parameters(self, param, variables):
+        """Replace any parameters passed as {} in the yaml file with the 
+        variable names that are passed in
+        
+        Only strings, lists of strings, and dictionaries of strings can have 
+        replaceable values at the moment
+        """
+        if type(param).__name__ == 'str': 
```

Comment:
These explicit type checks will break on python 3 and on python 2 randomly if the string you get happens to be unicode.

I would check for dict and list and otherwise assume its a string and template it.